src/libs/networks/networks.ts

```diff
@@ -112,7 +131,9 @@ export async function getNetworkInfo(
   }
 
   let flagged = false
-  const provider = getRpcProvider([rpcUrl], chainId)
+  const provider = getRpcProvider([rpcUrl], chainId, undefined, {
+    batchMaxCount: network ? getProviderBatchMaxCount(network) : 1
```

Comment:
We're passing network to getProviderBatchMaxCount, which in turns checks network.selectedRpcUrl, which is wrong as we want to know info about the rpcUrl that we're going to use from now on